### ISSUE TITLE
Which version of this repo is deployed on production? Need to reproduce an issue

### ISSUE THREAD

**Issue by timaschew** [houndci/eslint#29]
I tried to reproduce error which are happening and described here: https://github.com/houndci/eslint/issues/27

So I setup this repo on my local machine, after fixing an errror (#28) it worked (lint and show errors), what on the production hound is not working (tells that there are not linting errors). 
How can I reproduce the errors on my local machine?

**Comment by teoljungberg** [comment]
The previous revision that was deployed was 1f53e32, I redeployed master just now so production is on-par with master.

**Comment by timaschew** [comment]
I think I got it.
Seems that Hound is do something like a diff between the maser and the pull request branch.
And if there are already violations in the master it will not comment the old violations (or maybe no violations at all).
So it has nothing to do with the `.eslintrc` Maybe worth to tell this to the user in the docs. This issue took me several hours :)

Here are some examples:
not working: https://github.com/timaschew/hound-test/pull/8
working: https://github.com/timaschew/hound-test/pull/9

**Comment by teoljungberg** [comment]
@jcmorrow I think this might be worth looking into, do you mind looking into this today?

**Comment by jcmorrow** [comment]
I can definitely start investigating. I don't know how far I'll get but I'll update this thread by EOD.

**Comment by jcmorrow** [comment]
@timaschew I *think* I understand your question, but let me rephrase it just to make sure:

You want to know why hound doesn't comment on lines which

 1. Contain violations
 1. Aren't changed in the commit (in other words, they already exist in master)

This is actually the intended behavior of hound. At thoughtbot, we don't believe in rewriting existing code to conform to style guidelines. Therefore, hound only comments on violations in lines that have changed. From the FAQ:

> Can I have Hound check my entire repository for style?

>No, and we don't recommend doing so. Our rule is "Don't rewrite existing code to follow the style guide." See the thoughtbot style guide.

>It can be dangerous to re-write a lot of code for just style reasons. It can introduce functional regressions, which are a higher-level concern than style.

>We believe the right approach is to convert a codebase over time, one pull request at a time, to conform to a style guide. Only focus on style in diffs and make smaller changes that ideally have tests associated with them.

>If you'd still like to analyze your whole repo, try using Rubocop locally.

If this *isn't* your question, I'll be happy to re-investigate 😄

**Comment by timaschew** [comment]
Oh, okay you're right!
Of course this makes sense ;)
Thank you for clarification!

**Comment by teoljungberg** [comment]
Sounds like this is resolved, please re-open the issue if that's not the case.
